azavea/pwd-fishway
# Add font files and helper function for @font-face

## Overview
Adds typeface via `@font-face`, so that we're not depending on Google Fonts. 

Connects #46

### Demo
<img width="969" alt="Screen Shot 2019-05-03 at 12 20 09 PM" src="https://user-images.githubusercontent.com/5672295/57151012-d574b780-6d9d-11e9-992d-bbad88f237f6.png">
<img width="833" alt="Screen Shot 2019-05-03 at 12 20 37 PM" src="https://user-images.githubusercontent.com/5672295/57151025-e32a3d00-6d9d-11e9-9420-fc2f2c96b237.png">


### Notes
* I used a little helper function for `globalStyles.js` so that it's a bit easier to do `@font-face`.  We're currently only using weights 400, 500, 600, and 800. Italics are being used for all but 800.
* ~I'm not sure if there's a way to test that this works without internet... I tried turning off my wifi and the whole app didn't load.~ << lies, it works offline!

## Testing Instructions
* `git pull`
* You'll probably need to rerun the server
* You should see that the app is using the Fira font
* Test that font is loading without WiFi by disconnecting and refreshing the page
* Ensure that all fonts are loaded correctly by updating this declaration block:
Regular
```css
body {
       ...
       font-weight: 400;
    }
``` 
Regular Italic
```css
body {
       ...
       font-weight: 400;
       font-style: italic;
    }
``` 
Medium
```css
body {
       ...
       font-weight: 500;
    }
``` 
Medium Italic
```css
body {
       ...
       font-weight: 500;
       font-style: italic;
    }
``` 
Semibold
```css
body {
       ...
       font-weight: 600;
    }
``` 
Semibold Italic
```css
body {
       ...
       font-weight: 600;
       font-style: italic;
    }
``` 
Black
```css
body {
       ...
       font-weight: 800;
    }
```
